internal/pkg/skuba/kubernetes/pod.go

```diff
@@ -30,8 +32,23 @@ func getPodContainerImageTag(client clientset.Interface, namespace string, podNa
 	if err != nil {
 		return "", errors.Wrap(err, "could not retrieve pod object")
 	}
+	return getPodContainerImageTagFromPodObject(*podObject), nil
+}
+
+// return the image version tag without calling the api, just with the Pod object
+func getPodContainerImageTagFromPodObject(podObject v1.Pod) string {
 	containerImageWithName := podObject.Spec.Containers[0].Image
 	containerImageTag := strings.Split(containerImageWithName, ":")
 
-	return containerImageTag[len(containerImageTag)-1], nil
+	return containerImageTag[len(containerImageTag)-1]
+}
+
+// return a pod object matched from a podList
+func getPodFromPodList(list []v1.Pod, name string) (v1.Pod, error) {
+	for _, pod := range list {
+		if name == pod.Name {
+			return pod, nil
+		}
+	}
+	return v1.Pod{}, errors.New(fmt.Sprintf("Could not find pod %s in pod list", name))
```

Comment:
As [1], I would prefer is pod not exist, return `nil`.

Reply:
> As [1], I would prefer is pod not exist, return nil.

So for this, my first approach was to return `nil`, but as the func returns a `v1.Pod, error` it was complaining that I was returning a `nil` instead of a expected `v1.Pod`

Is this expected and safe to ignore?

Reply:
>  This line could be simply as fmt.Errorf("Could not find pod %s in pod list", name)

Nice, thanks!

Reply:
> > As [1], I would prefer is pod not exist, return nil.
> 
> So for this, my first approach was to return `nil`, but as the func returns a `v1.Pod, error` it was complaining that I was returning a `nil` instead of a expected `v1.Pod`

golint or golang-ci complaing?

Reply:
`make lint`

```
GO111MODULE=on go vet ./...
# github.com/SUSE/skuba/internal/pkg/skuba/kubernetes
internal/pkg/skuba/kubernetes/pod.go:53:2: cannot use nil as type "k8s.io/api/core/v1".Pod in return argument
make: *** [Makefile:94: lint] Error 2
```

Reply:
I saw the slice []v1.Pod is not a pointer.
Let's keep your original code, it seems the elegant way.

Reply:
oh yeah, I see. If I change it to return `*v1.Pod` I can return nil :)